csharp/test/Ice/ami/AllTests.cs

```diff
@@ -102,3172 +77,512 @@ public void Report(bool sentSynchronously)
                         test(false);
                     }
-                }
-
-                private Cookie _cookie;
-            }
-
-            private class ResponseCallback : CallbackBase
-            {
-                public ResponseCallback()
-                {
-                }
-
-                public void
-                isA(bool r)
-                {
-                    test(r);
-                    called();
-                }
-
-                public void
-                ping()
-                {
-                    called();
-                }
-
-                public void
-                id(string id)
-                {
-                    test(id.Equals("::Test::TestIntf"));
-                    called();
-                }
-
-                public void
-                ids(string[] ids)
-                {
-                    test(ids.Length == 2);
-                    called();
-                }
-
-                public void
-                connection(Ice.Connection conn)
-                {
-                    test(conn != null);
-                    called();
-                }
-
-                public void
-                op()
-                {
-                    called();
-                }
+                    catch(NoEndpointException)
+                    {
+                    }
 
-                public void
-                opWithResult(int r)
-                {
-                    test(r == 15);
-                    called();
-                }
+                    // Ensures no exception is called when response is received
+                    test(await p.ice_isAAsync("::Test::TestIntf"));
+                    await p.opAsync();
+                    await p.opWithResultAsync();
 
-                public void
-                opWithUE(Ice.Exception e)
-                {
+                    // If response is a user exception, it should be received.
                     try
                     {
-                        throw e;
+                        await p.opWithUEAsync();
+                        test(false);
                     }
                     catch(Test.TestIntfException)
                     {
-                        called();
-                    }
-                    catch(Ice.Exception)
-                    {
-                        test(false);
                     }
                 }
+                output.WriteLine("ok");
 
-            }
-
-            private class ExceptionCallback : CallbackBase
-            {
-                public ExceptionCallback()
-                {
-                }
-
-                public void
-                isA(bool r)
+                output.Write("testing progress callback... ");
+                output.Flush();
                 {
-                    test(false);
-                }
+                    {
+                        var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
It's not immediately clear what we're testing here. Seems that we verify progress is called - but then why retest this many times?

Reply:
I think it is because the builtin operations are manually implemented in Proxy.cs, so we test with all the builtin + with one generated operation

Reply:
ok